Ogre installation error:  incomplete definition of type 'std::char_traits<unsigned short>'

HI, I am trying to install `gz-harmonic`. However, I have been facing this error and I cannot proceed the installation of `gz-harmonic`. 
</br>

Here is my `brew config`
~~~
$ brew config
HOMEBREW_VERSION: 4.4.30-2-g2664c05
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2664c0568dc854e498ca257b8484a07fcfb0eae0
Last commit: 3 hours ago
Branch: master
Core tap JSON: 14 Apr 17:30 UTC
Core cask tap JSON: 14 Apr 17:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.wglgVBsMi9/org.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 11
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.8/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.4-arm64
CLT: 16.3.0.0.1.1742442376
Xcode: 16.0
Rosetta 2: false
~~~

This is the build log
~~~
brew install ogre2.3 --verbose

...

[ 41%] Building CXX object OgreMain/CMakeFiles/OgreMain.dir/src/OgreVertexShadowMapHelper.cpp.o
cd /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/build/OgreMain && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DFREEIMAGE_LIB -DOGRE_NONCLIENT_BUILD -DOgreMain_EXPORTS -D_MT -D_USRDLL -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/build/include -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OSX -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/RenderSystems/Metal/include/Windowing/OSX -I/usr/local/include -I/usr/local/include/freetype2 -F/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1 -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/Threading -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/nedmalloc -Wall -Winit-self -Wno-overloaded-virtual -Wcast-qual -Wwrite-strings -Wextra -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long  -msse -msse2 -O3 -DNDEBUG -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.0.sdk -fPIC -DOGRE_GCC_VISIBILITY -fvisibility=hidden -fvisibility-inlines-hidden -MD -MT OgreMain/CMakeFiles/OgreMain.dir/src/OgreVertexShadowMapHelper.cpp.o -MF CMakeFiles/OgreMain.dir/src/OgreVertexShadowMapHelper.cpp.o.d -o CMakeFiles/OgreMain.dir/src/OgreVertexShadowMapHelper.cpp.o -c /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreVertexShadowMapHelper.cpp
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:821:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
  821 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreUTFString.h:235:13: note: in instantiation of template class 'std::basic_string<unsigned short>' requested here
  235 |             dstring::iterator mIter;
      |             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:648:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string_view:300:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
  300 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__type_traits/is_convertible.h:22:102: note: in instantiation of template class 'std::basic_string_view<unsigned short>' requested here
   22 | struct _LIBCPP_TEMPLATE_VIS is_convertible : public integral_constant<bool, __is_convertible(_T1, _T2)> {};
      |                                                                                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:746:29: note: in instantiation of template class 'std::is_convertible<const unsigned short *const &, std::basic_string_view<unsigned short>>' requested here
  746 |     : public _BoolConstant< is_convertible<const _Tp&, basic_string_view<_CharT, _Traits> >::value &&
      |                             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:1543:38: note: in instantiation of template class 'std::__can_be_converted_to_string_view<unsigned short, std::char_traits<unsigned short>, const unsigned short *>' requested here
 1543 |   template <class _Tp, __enable_if_t<__can_be_converted_to_string_view<_CharT, _Traits, _Tp>::value, int> = 0>
      |                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:1545:3: note: while substituting prior template arguments into non-type template parameter [with _Tp = const code_point *]
 1545 |   insert(size_type __pos1, const _Tp& __t) {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1546 |     __self_view __sv = __t;
      |     ~~~~~~~~~~~~~~~~~~~~~~~
 1547 |     return insert(__pos1, __sv.data(), __sv.size());
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1548 |   }
      |   ~
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreUTFString.h:662:19: note: while substituting deduced template arguments into function template 'insert' [with _Tp = const code_point *, $1 = (no value)]
  662 |             mData.insert( index, str );
      |                   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:821:42: error: implicit instantiation of undefined template 'std::char_traits<unsigned int>'
  821 |   static_assert(is_same<_CharT, typename traits_type::char_type>::value,
      |                                          ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:745:39: note: in instantiation of template class 'std::basic_string<unsigned int>' requested here
  745 |         return mBuffer.mUTF32StrBuffer->c_str();
      |                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3088:29: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 3088 |   return insert(__pos, __s, traits_type::length(__s));
      |                             ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreUTFString.h:662:19: note: in instantiation of member function 'std::basic_string<unsigned short>::insert' requested here
  662 |             mData.insert( index, str );
      |                   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | st[ 41%] Building CXX object OgreMain/CMakeFiles/OgreMain.dir/src/OgreViewport.cpp.o
cd /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/build/OgreMain && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DFREEIMAGE_LIB -DOGRE_NONCLIENT_BUILD -DOgreMain_EXPORTS -D_MT -D_USRDLL -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/build/include -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OSX -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/RenderSystems/Metal/include/Windowing/OSX -I/usr/local/include -I/usr/local/include/freetype2 -F/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1 -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/Threading -I/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/nedmalloc -Wall -Winit-self -Wno-overloaded-virtual -Wcast-qual -Wwrite-strings -Wextra -Wno-unused-parameter -Wshadow -Wno-missing-field-initializers -Wno-long-long  -msse -msse2 -O3 -DNDEBUG -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.0.sdk -fPIC -DOGRE_GCC_VISIBILITY -fvisibility=hidden -fvisibility-inlines-hidden -MD -MT OgreMain/CMakeFiles/OgreMain.dir/src/OgreViewport.cpp.o -MF CMakeFiles/OgreMain.dir/src/OgreViewport.cpp.o.d -o CMakeFiles/OgreMain.dir/src/OgreViewport.cpp.o -c /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreViewport.cpp
ruct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3088:40: error: incomplete definition of type 'std::char_traits<unsigned short>'
 3088 |   return insert(__pos, __s, traits_type::length(__s));
      |                             ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2576:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 2576 |     traits_type::copy(std::__to_address(__p), __s, __n);
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2663:16: note: in instantiation of function template specialization 'std::basic_string<unsigned short>::__assign_no_alias<true>' requested here
 2663 |         return __assign_no_alias<true>(__str.data(), __str.size());
      |                ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:533:15: note: in instantiation of member function 'std::basic_string<unsigned short>::operator=' requested here
  533 |         mData = copy.mData;
      |               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2576:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 2576 |     traits_type::copy(std::__to_address(__p), __s, __n);
      |     ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2577:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 2577 |     traits_type::assign(__p[__n], value_type());
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2577:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 2577 |     traits_type::assign(__p[__n], value_type());
      |     ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2576:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 2576 |     traits_type::copy(std::__to_address(__p), __s, __n);
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2666:14: note: in instantiation of function template specialization 'std::basic_string<unsigned short>::__assign_no_alias<false>' requested here
 2666 |       return __assign_no_alias<false>(__str.data(), __str.size());
      |              ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:533:15: note: in instantiation of member function 'std::basic_string<unsigned short>::operator=' requested here
  533 |         mData = copy.mData;
      |               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2576:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 2576 |     traits_type::copy(std::__to_address(__p), __s, __n);
      |     ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2577:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 2577 |     traits_type::assign(__p[__n], value_type());
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2577:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 2577 |     traits_type::assign(__p[__n], value_type());
      |     ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3397:3: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 3397 |   traits_type::copy(std::__to_address(__new_data), std::__to_address(__p), size() + 1);
      |   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3334:3: note: in instantiation of member function 'std::basic_string<unsigned short>::__shrink_or_extend' requested here
 3334 |   __shrink_or_extend(__target_capacity);
      |   ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:639:15: note: in instantiation of member function 'std::basic_string<unsigned short>::reserve' requested here
  639 |         mData.reserve( inSize );
      |               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3397:14: error: incomplete definition of type 'std::char_traits<unsigned short>'
 3397 |   traits_type::copy(std::__to_address(__new_data), std::__to_address(__p), size() + 1);
      |   ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2222:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 2222 |     traits_type::assign(__p[__newsz], value_type());
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2135:5: note: in instantiation of member function 'std::basic_string<unsigned short>::__null_terminate_at' requested here
 2135 |     __null_terminate_at(std::__to_address(__get_pointer()), __pos);
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3305:5: note: in instantiation of member function 'std::basic_string<unsigned short>::__erase_to_end' requested here
 3305 |     __erase_to_end(__n);
      |     ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:644:15: note: in instantiation of member function 'std::basic_string<unsigned short>::resize' requested here
  644 |         mData.resize( num, val );
      |               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:2222:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 2222 |     traits_type::assign(__p[__newsz], value_type());
      |     ~~~~~~~~~~~^~
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3290:5: error: implicit instantiation of undefined template 'std::char_traits<unsigned short>'
 3290 |     traits_type::assign(*__get_long_pointer(), value_type());
      |     ^
/private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:674:15: note: in instantiation of member function 'std::basic_string<unsigned short>::clear' requested here
  674 |         mData.clear();
      |               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
   23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
      |                             ^
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/src/OgreUTFString.cpp:28:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreStableHeaders.h:44:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgreCommon.h:32:
In file included from /private/tmp/ogre2.3-20250414-57853-nmprwt/ogre-next-2.3.1/OgreMain/include/OgrePrerequisites.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include/c++/v1/string:3290:16: error: incomplete definition of type 'std::char_traits<unsigned short>'
 3290 |     traits_type::assign(*__get_long_pointer(), value_type());
      |     ~~~~~~~~~~~^~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [OgreMain/CMakeFiles/OgreMain.dir/src/OgreUTFString.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [OgreMain/CMakeFiles/OgreMain.dir/all] Error 2
make: *** [all] Error 2

HOMEBREW_VERSION: 4.4.30-2-g2664c05
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2664c0568dc854e498ca257b8484a07fcfb0eae0
Last commit: 3 hours ago
Branch: master
Core tap JSON: 14 Apr 17:30 UTC
Core cask tap JSON: 14 Apr 17:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.wglgVBsMi9/org.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 11
HOMEBREW_NO_BOOTSNAP: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.8/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.4-arm64
CLT: 16.3.0.0.1.1742442376
Xcode: 16.0
Rosetta 2: false

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j11
CMAKE_PREFIX_PATH: /opt/homebrew
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_PATH: /opt/homebrew/opt/libpng/lib/pkgconfig:/opt/homebrew/opt/freetype/lib/pkgconfig:/opt/homebrew/opt/libxau/lib/pkgconfig:/opt/homebrew/opt/libxdmcp/lib/pkgconfig:/opt/homebrew/opt/libxcb/lib/pkgconfig:/opt/homebrew/opt/libx11/lib/pkgconfig:/opt/homebrew/opt/libzzip/lib/pkgconfig:/opt/homebrew/opt/rapidjson/lib/pkgconfig:/opt/homebrew/opt/hwloc/lib/pkgconfig:/opt/homebrew/opt/tbb/lib/pkgconfig:/opt/homebrew/opt/xorgproto/share/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/15
HOMEBREW_MAKE_JOBS: 11
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk
ACLOCAL_PATH: /opt/homebrew/share/aclocal
PATH: /opt/homebrew/Library/Homebrew/shims/mac/super:/opt/homebrew/opt/cmake/bin:/opt/homebrew/opt/doxygen/bin:/opt/homebrew/opt/libpng/bin:/opt/homebrew/opt/freetype/bin:/opt/homebrew/opt/libzzip/bin:/opt/homebrew/opt/hwloc/bin:/usr/bin:/bin:/usr/sbin:/sbin

~~~

Has anyone solved this issue on MacOS 15 with M3 cpu?

--- a/Formula/ogre2.3.rb
+++ b/Formula/ogre2.3.rb
@@ -28,6 +28,12 @@ class Ogre23 < Formula
   depends_on "rapidjson"
   depends_on "tbb"
 
+  patch do
+    # Fix for compatibility with XCode 16.3
+    url "https://github.com/scpeters/ogre-next/commit/b7439ae047489aa104a6775a99a9e93294c3d5b5.patch?full_index=1"
+    sha256 "d56016cd237c9a98e7c4389c57a455ea5d660d538d0cb1d5082bb2f9ed4e00b8"
+  end
+
   def install
     rpaths = [
       rpath,

